tests/virt_autotest/virt_utils.pm

```diff
@@ -226,17 +226,35 @@ sub handle_sp_in_settings {
 
     # We need small case variable value
     my $var_value = lc(get_required_var("$var_name"));
-    # Add $value_for_sp after release for products like sle15, sle16
-    if ($var_value !~ /sp|fcs/ && $value_for_sp) {
-        $var_value =~ s/(sles-\d+)/$1-$value_for_sp/;
+
+    my ($major_version) = $var_value =~ /sles-(\d+)/;
+    if ($major_version >= 16) {
+        # Follow SLES16+ Major.Minor format
+        if ($var_value !~ /sles-\d+[\.-]\d+/) {
+            $var_value .= "-0";
+        }
+    }
+    else {
+        # Add $value_for_sp after release for products like sle12, sle15
+        if ($var_value !~ /sp|fcs/ && $value_for_sp) {
+            $var_value =~ s/(sles-\d+)/$1-$value_for_sp/;
+        }
```

Comment:
If you frequently use > or < 16, I think you'd better have a subroutine like is_pre_16 or something. There is already is_pre_15 in lib/version_utils.pm.